### ISSUE TITLE
Unexpected error occured when running migrate up in latest alpha.

### ISSUE THREAD

**Issue by AlexanderProd** [prisma/prisma2#1423]
When running `npx prisma2 migrate up --experimental` in the latest alpha `2.0.0-alpha.618` there's an unexpected error. 

```
Checking the datasource for potential data loss...
ERROR  Oops, an unexpected error occured!
```

OS: macOS
Prisma2 version: 2.0.0-alpha.618 (using alpha because of #1419 )
DB: MySQL
Report id: 1354

Schema:
```
datasource mysql {
  url      = "mysql://******:*****@******/*****"
  provider = "mysql"
}

generator client {
  provider = "prisma-client-js"
}

model User {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  email     String   @unique
  name      String?
  phone     String?
  role      Role     @default(USER)
  posts     Post[]
}

model Post {
  id         String     @id @default(cuid())
  createdAt  DateTime   @default(now())
  updatedAt  DateTime   @updatedAt
  author     User
  title      String
  categories Category[]
  published  Boolean    @default(false)
}

model Category {
  id    String  @id @default(cuid())
  name  String 
}

enum Role {
  USER
  ADMIN
}
```

those are the changes that would've been applied: 
<img width="441" alt="image" src="https://user-images.githubusercontent.com/3080903/72978715-3f733800-3dd7-11ea-9a5a-d7501c177af7.png">

**Comment by timsuchanek** [comment]
@AlexanderProd, unfortunately, we can't reproduce this.

You mention that you used `npx prisma2 migrate up --experimental` - that however would run the "latest" version from npm, not the "alpha" version. Was it just a typo or did you use `npx prisma2@alpha migrate up --experimental`?

**Comment by AlexanderProd** [comment]
@timsuchanek  Yeah I did indeed run it using npx (though running with `npx` uses the version installed in my local `node_modules`), but I also tried it without `npx` using the global version (Installed the alpha globaly too). The error occurs in both cases.

**Comment by janpio** [comment]
(Using `npx` with the alpha in `package.json` should indeed use that alpha version afaik)

**Comment by AlexanderProd** [comment]
The alpha is indeed in my `package.json`.

`"prisma2": "^2.0.0-alpha.618",`

`prisma2 migrate save --experimental` wouldn't have worked otherwise because of #1419 .

**Comment by janpio** [comment]
(You can also confirm with `npx prisma2 -v`)

**Comment by AlexanderProd** [comment]
@janpio Thanks, this is the exact version I'm using 
`prisma2@2.0.0-alpha.618, binary version: 0150d2c09b7ddee6b866a924dea38ba157ad4980`

**Comment by AhmedElywa** [comment]
@AlexanderProd  i get same error like you when you use version like this `^2.0.0-alpha.618` with `^` it's get latest version and you can see this in yarn.lock file

**Comment by AlexanderProd** [comment]
I guess the problem originates from my MySQL (Maria DB) version. 

I've tried running the following generated SQL script inside the `migrations` folder and I get an error. 

```
CREATE TABLE `521_prisma-test`.`Category` (
    `id` varchar(191) NOT NULL  ,
    `name` varchar(191) NOT NULL DEFAULT '' ,
    `post` varchar(191)  ,
    PRIMARY KEY (`id`)
) 
DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci

ALTER TABLE `521_prisma-test`.`Category` ADD FOREIGN KEY (`post`) REFERENCES `521_prisma-test`.`Post`(`id`) ON DELETE SET NULL
```

This is the error I'm getting when running the SQL myself. 

```
You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'ALTER TABLE `521_prisma-test`.`Category` ADD FOREIGN KEY (`post`) REFERENCES `52' at line 9
```

I'm using MySQL 5.5.5 (MariaDB 10.0.38)

I'm not sure though if this is because im using a MySQL (MariaDB) version that isn't supported by Prisma or if prisma made a mistake generating the correct script for this version.

**Comment by tomhoule** [comment]
@AlexanderProd I think the problem here could be that the database name starts with a number. I'll try to confirm this in tests.

**Comment by tomhoule** [comment]
I just confirmed it's not because of the digits in the database name.

**Comment by AlexanderProd** [comment]
I've can also successfully run the generated SQL query by doing it step by step (first creating the new table and then altering it). So the SQL is fine.
